tests/manual/test_memory_reporting.py

```
@@ -0,0 +1,293 @@
+        with open(tmp_path, "wb") as f:
+            for chunk in r.iter_content(chunk_size=1024 * 1024):
+                if chunk:
+                    f.write(chunk)
+    os.replace(tmp_path, cached_path)
+    print(f"Cached snapshot to {cached_path} ({cached_path.stat().st_size} bytes)")
+    return cached_path
+
+
+def get_allocated_bytes() -> int:
+    """Get jemalloc allocated_bytes from telemetry."""
+    resp = requests.get(f"{BASE_URL}/telemetry", params={"details_level": 2})
```

Ruff (0.15.9)

[error] Probable use of `requests` call without timeout

(S113)